File: app/src/main/java/com/example/admin/smarttranslator/Models/PhotoCard.java
package com.example.admin.smarttranslator.Models;

import android.graphics.Bitmap;
import java.util.ArrayList;
import java.util.List;

public class PhotoCard {
    private int id;
    private String filePath;
    private boolean liked = false;
    private String lanFrom;
    private String lanTo;
    private List<String> resultFrom = new ArrayList<>();
    private List<String> resultTo = new ArrayList<>();
    private List<String> neoResult;
    private Bitmap bitmap;

    public static int width;

    public PhotoCard(){}

    public PhotoCard(String path, String lanFrom, String lanTo, List<String> neoResult , List<String> resultFrom, List<String> resultTo){
        filePath = path;
        this.lanFrom = lanFrom;
        this.lanTo = lanTo;
        this.neoResult = neoResult;
        this.resultFrom = resultFrom;
        this.resultTo = resultTo;
    }

    public PhotoCard(String lanFrom, String lanTo){
        this.lanFrom = lanFrom;
        this.lanTo = lanTo;
    }

    public Bitmap getBitmap(){ return bitmap; }

    public void setBitmap(Bitmap bitmap){ this.bitmap = bitmap; }

    public String getFilePath(){return filePath;}

    public void setFilePath(String path){ filePath = path; }

    public PhotoCard(String lan){
        lanFrom = lan;
    }

    public boolean isLiked() {
        return liked;
    }

    public void setLiked(boolean liked){ this.liked = liked; }

    public void changeLike(){ liked = !liked; }

    public String getLanFrom() {
        return lanFrom;
    }

    public void setLanFrom(String lanFrom) {
        this.lanFrom = lanFrom;
    }

    public String getLanTo() {
        return lanTo;
    }

    public void setLanTo(String lanTo) {
        this.lanTo = lanTo;
    }

    public List<String> getResultFrom() {
        return resultFrom;
    }

    public void setResultFrom(List<String> resultFrom) {
        this.resultFrom = resultFrom;
    }

    public List<String> getResultTo() {
        return resultTo;
    }

    public void setResultTo(List<String> resultTo) {
        this.resultTo = resultTo;
    }

    public List<String> getNeoResult() {
        return neoResult;
    }

    public void setNeoResult(List<String> result){ neoResult = result; }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: app/src/main/java/com/example/admin/smarttranslator/Models/PhotoCard.kt
package com.example.admin.smarttranslator.Models

import android.graphics.Bitmap
import java.util.ArrayList

class PhotoCard {
    var id: Int = 0
    var filePath: String? = null
    var isLiked = false
    var lanFrom: String? = null
    var lanTo: String? = null
    var resultFrom: List<String> = ArrayList()
    var resultTo: List<String> = ArrayList()
    var neoResult: List<String>? = null
    var bitmap: Bitmap? = null

    constructor() {}

    constructor(path: String, lanFrom: String, lanTo: String, neoResult: List<String>, resultFrom: List<String>, resultTo: List<String>) {
        filePath = path
        this.lanFrom = lanFrom
        this.lanTo = lanTo
        this.neoResult = neoResult
        this.resultFrom = resultFrom
        this.resultTo = resultTo
    }

    constructor(lanFrom: String, lanTo: String) {
        this.lanFrom = lanFrom
        this.lanTo = lanTo
    }

    constructor(lan: String) {
        lanFrom = lan
    }

    fun changeLike() {
        isLiked = !isLiked
    }

    companion object {

        var width: Int = 0
    }
}
